### ISSUE TITLE
AUR package

### ISSUE THREAD

**Issue by wezm** [daa84/neovim-gtk#13]
FYI I made a first pass at an AUR package for neovim-gtk: <https://aur.archlinux.org/packages/neovim-gtk-git>. This makes it easy to build and install on Arch Linux:

```
git clone https://aur.archlinux.org/neovim-gtk-git.git
cd neovim-gtk-git 
makepkg -si
```

**Comment by daa84** [comment]
Thank you 😄 
I'm not user of Arch so can't test it, but think I can put this instructions to README 👍 
thanks

**Comment by wezm** [comment]
I'm basing on the makedepends on what's done for the [ripgrep](https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/ripgrep), which is a rust tool in the community repos. It depends on cargo, which is specified as a package provided by [rust](https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/rust#n47).

**Comment by daa84** [comment]
Backtrace looks like some bug
But I can check this only after about 2 weeks

**Comment by daa84** [comment]
Can't reproduce, but think it must be fixed :smile:

**Comment by codedmart** [comment]
I just tried to use the AUR package and am getting this error: `error: no default toolchain configured`.

**Comment by wezm** [comment]
It sounds like you have rustup installed without a default toolchain set. I had wondered about this. At the moment it think the PKGBUILD is using whatever `cargo` it finds in your path rather than the Arch packaged one (that is installed as part of the dependency on rust). I'll update the PKGBUILD to use absolute paths and that should fix it.

**Comment by wezm** [comment]
I pushed a revised `PKGBUILD` that uses the absolute path to `cargo`, give that a try and see if that helps.

**Comment by davidsierradz** [comment]
I have both `neovim-qt` and `neovim-gtk` AUR packages installed, I installed `neovim-qt` first then later on `neovim-gtk`, checking around this repo files I noticed that `nvim_gui_shim.vim` its a file shared by both projects, when I run `:scriptnames` from `neovim-gtk` I get that `/usr/share/nvim/runtime/plugin/nvim_gui_shim.vim` file sourced, but its the `neovim-qt` version not the `neovim-gtk` one.

**Comment by wezm** [comment]
Interesting. I suppose the NeovimGtk one should have a unique name. I wonder what the consequence of having two versions of the same shim in the runtime path. I suppose alternatively we could create a nvim-gui-shim package that both UIs depend upon?

**Comment by davidsierradz** [comment]
[This one](https://github.com/equalsraf/neovim-gui-shim) by the `neovim-qt` author seems to be.

**Comment by daa84** [comment]
if call simple `make install` it will be placed at `$(PREFIX)/share/nvim-gtk/ `
https://github.com/daa84/neovim-gtk/blob/a6e16d1cbdb64031855612ae25df31592cdee249/Makefile#L14
and this path added to rtp at start
There is also setting that can change this path after installation https://github.com/daa84/neovim-gtk/wiki/Configuration#runtime

**Comment by wezm** [comment]
Ok I'll have a look into changing the package to use or mirror the Makefile behviour.

**Comment by wezm** [comment]
I updated the `PKGBUILD` to better mirror the Makefile.

**Comment by davidsierradz** [comment]
Thanks, the file now it is installed on `/usr/share/nvim-gtk/runtime/nvim_gui_shim.vim` but its not sourced, according to `help 'runtimepath'`:

```
	4. In nvim/site subdirectories for each directory in $XDG_DATA_DIRS.  
	   This is for plugins which were installed by system administrator, 
	   but are not part of the Neovim distribution. XDG_DATA_DIRS defaults 
	   to /usr/local/share/:/usr/share/, so system administrators are 
	   expected to install site plugins to /usr/share/nvim/site.
```

So it should be installed in `/usr/share/nvim/nvim-gtk`?
